franklin/github/views.py

```diff
@@ -130,14 +132,50 @@ def register_repo(request):
                 return Response(message, status=status.HTTP_403_FORBIDDEN)
     return Response(status=status.HTTP_400_BAD_REQUEST)
 
+
 @api_view(['GET'])
-def get_user_repos(request):
+def deployed_repos(request):
+    """
+    All repos currently deployed by Franklin that the user can manage.
+    ---
+
+    response_serializer: SiteSerializer
+
+    responseMessages:
+        - code: 500
+          message: Error from Github.
+    """
     if request.method == 'GET':
         if request.user.details.sites.count() == 0:
-            request.user.details.update_repos_for_user()
+            github_repos = request.user.details.get_user_repos()
+            # TODO - return error from github
+            request.user.details.update_repos_for_user(github_repos)
         site_data = request.user.details.sites.all()
         serializer = SiteSerializer(site_data, many=True)
-        return Response(serializer.data)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+@api_view(['GET'])
+def deployable_repos(request):
+    """
+    All repos from Github that the user has the permission level to deploy
+    ---
+
+    responseMessages:
+        - code: 500
+          message: Error from Github.
+    """
+    if request.method == 'GET':
+        # TODO - in the model, github response should map to a serializer which
+        # we should use here to define the respone type
+        github_repos = request.user.details.get_user_repos()
+        # TODO - return error from github
```

Comment:
Was tempted to fix these TODO items now, but that would take me down the path of refactoring a lot of things. Would prefer to wait until the last week before the holiday to kill a lot of these lingering code quality items.